lib/device/char.lua

```diff
@@ -0,0 +1,30 @@
+--
+-- SPDX-FileCopyrightText: (c) 2023-2025 Ring Zero Desenvolvimento de Software LTDA
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+local device = require("device")
+local chardev = {}
+
+-- No-operation function to use as default for open and release
+local function nop() end
+
+-- Create a new character device with default values for open and release
+function chardev.new(t)
+    -- If t is a string, convert it to a table with name field
+    if type(t) == "string" then
+        t = {name = t}
+    end
+    
+    -- Ensure t is a table
+    t = t or {}
+    
+    -- Set default values for open and release if not provided
```

Comment:
why only open and release? perhaps it would be better to have a table with all file operations..